src/components/pages/DashboardPage.js

```diff
@@ -14,16 +14,17 @@ export default class DashboardPage extends Component {
   setStepTo1 = () => this.setState({ mainStep: 1 });
   setStepTo2 = () => this.setState({ mainStep: 2 });
   setStepTo3 = () => this.setState({ mainStep: 3 });
+  setStepTo0 = () => this.setState({ mainStep: 0 });
```

Comment:
Try and re-use more of your code instead of hardcoding the step numbers like you do in the setSeptTo functions here.
You could avoid all this repetition by passing in the step number to a re-usable function, for example: `changeStep(n)`? 
So on line 60 where the onClick event is, you could just say something like `onClick={this.changeStep(3)}`, and then only show step 3 if `this.state.mainStep === 3`
This will require changes across several components as the props would change, but it's good practice to do this. Imagine if you had 60 different steps - would you have a function for each one?